crates/js-component-bindgen/src/transpile_bindgen.rs

```diff
@@ -294,8 +294,31 @@ impl<'a> JsBindgen<'a> {
             uwrite!(
                 output,
                 "\
-                        {}\
-                        {};
+                        let gen = (function* init () {{
+                            {}\
+                            {};
+                        }})();
+                        let promise, resolve, reject;
+                        function runNext (value) {{
+                            try {{
+                                let done;
+                                do {{
+                                    ({{ value, done }} = gen.next(value));
+                                }} while (!(value instanceof Promise) && !done);
+                                if (done) {{
+                                    if (resolve) resolve(value);
+                                    else return value;
```

Comment:
This `return` doesn't go anywhere: one of the two calls to `runNext` only happens in the `if (resolve)` case, and the other is thrown away rather than being returned.

I can't tell if this block of code is meant to support both sync and async instantiation. If it's just sync, you don't need to worry about promises, and can drive the generator much more simply:

```js
let done, value;
while (!done) {
  0, { done, value } = gen.next(value);
}
return value;
```

Reply:
That was an oversight, fixed.